yksmp/src/lib.rs

```diff
@@ -0,0 +1,375 @@
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::convert::TryInto;
+use std::ffi::c_void;
+use std::{ptr, slice};
+
+/// Parses LLVM stackmaps version 3 from a given address. Provides a way to query relevant
+/// locations given the return address of a `__llvm_deoptimize` function.
+/// Note that LLVM currently only supports stackmaps for 64 bit architectures. Once they support
```

Comment:
This is a very important thing to note in a minor comment! I think this comment should be a `//!` comment at the top of the module and we *must* add a compile-time guard that prevents this compiling on architectures we don't know for sure that it works on. In practise I think we must say "this only works on x86" as I don't think it's even true that it will work on all 64-bit platforms.

Reply:
LLVM stackmaps support X86_64, PowerPC, Aarch64 and SystemZ. But since I have no idea if this code runs on those, a guard sounds like a good idea.

Reply:
In an ideal world, as and when we test a new platform we can relax the compile-time guard.

Reply:
Can I just put `#[cfg(target_arch = "x86_64")]` at the top of the file or would that just guard the first import?

Reply:
Dunno.

Reply:
Fixed in 0f3997f.